Assignment4/adgiz05-22a334/task07.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1gJNRQugJJvhg3Mny3qdgy4oaP-RvYBfs
+
+**Task 07: Querying RDF(s)**
+"""
+
+#!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+for s, p, o in g:
+  print(s, p, o)
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+
+###RDFLib###
+print('RDFLib results:')
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s)
+
+###SPARQL###
+print('SPARQL results:')
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf+ ns:Person
+  }
+  ''',
+  initNs = {"ns": ns}
+)
+
+for r in g.query(q1):
+ print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+###RDFLib###
+print('RDFLib results:')
+
+for s1,p1,o1 in g.triples((None, RDF.type, ns.Person)):
+  print(s1)
+
+for s1,p1,o1 in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s2,p2,o2 in g.triples((None, RDF.type, s1)):
+    print(s2)
+
+###SPARQL###
+print('SPARQL results:')
+q1 = prepareQuery('''
+    SELECT ?Subject WHERE 
+    {
+      ?subclasses a|rdfs:subClassOf* ns:Person.
```

Comment:
this works, but it's not correct. You are doing an OR here. Ideally if you do a/rdfs:subClassOf* then you will obtain the right results without line 71